docs/resources/Template.md

```diff
@@ -112,3 +112,37 @@ Create a new guild based on a template. Returns a [guild](#DOCS_RESOURCES_GUILD/
 |-------|------------------------------------------|-----------------------------------------|
 | name  | string                                   | name of the guild (2-100 characters)    |
 | icon? | [image data](#DOCS_REFERENCE/image-data) | base64 128x128 image for the guild icon |
+
+## Get Guild Template % GET /guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-object}/templates
+
+Returns an array with a [template](#DOCS_RESOURCES_TEMPLATE/template-object) object for the guild's template, or an empty array if it doesn't have one.  Requires the `MANAGE_GUILD` permission.
+
+## Create Guild Template % POST /guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-object}/templates
+
+Creates a template for the guild. Requires the `MANAGE_GUILD` permission. Returns the created [template](#DOCS_RESOURCES_TEMPLATE/template-object) object on success.
+
+###### JSON Params
+
+| Field        | Type    | Description                                     |
+|--------------|---------|-------------------------------------------------|
+| name         | string  | name of the template (1-100 characters)         |
+| description? | ?string | description for the template (0-120 characters) |
+
+## Sync Guild Template % PUT /guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-object}/templates/{template.code#DOCS_RESOURCES_TEMPLATE/template-object}
+
+Syncs the template to the guild's current state. Requires the `MANAGE_GUILD` permission. Returns the [template](#DOCS_RESOURCES_TEMPLATE/template-object) object on success.
+
+## Modify Guild Template % PATCH /guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-object}/templates/{template.code#DOCS_RESOURCES_TEMPLATE/template-object}
+
+Modifies the template's metadata. Requires the `MANAGE_GUILD` permission. Returns the [template](#DOCS_RESOURCES_TEMPLATE/template-object) object on success.
+
+###### JSON Params
+
+| Field        | Type    | Description                                     |
+|--------------|---------|-------------------------------------------------|
+| name         | string  | name of the template (1-100 characters)         |
```

Comment:
Since this is a patch request, the name is also optional, right?

Reply:
it isn't:
![ss](https://advaith.is-in-hyper.space/473ac82315.png)

Reply:
That makes no sense for a patch request... What if I want to modify just the description?

Reply:
This is a good point. I'll look into changing this, along with the other templates-related issue.

Reply:
We should now accept both fields optionally (it is worth noting that description is still both nullable and optional.)

```suggestion
| name?        | string  | name of the template (1-100 characters)         |
```